lib/run-compilation.js

```diff
@@ -14,6 +14,7 @@ var errorMessage = require('./constants').errorMessage;
 module.exports = function runCompilation (options, compiler, done) {
   var errors = [];
   var warnings = [];
+  var compilerMethod = options.emitErrors ? 'after-compile' : 'after-emit';
```

Comment:
Is there some way we can write a new test for this branch? I _think_ the existing tests do cover it, which is why I'm okay with this change, but I'm not _completely_ sure.